client/boot/common.js

```diff
@@ -127,6 +127,10 @@ const oauthTokenMiddleware = () => {
 			'/connect',
 		];
 
+		if ( isJetpackCloud() && ! isJetpackCloudProd() ) {
```

Comment:
I'm not a fan of adding an extra environment condition like this, but I think that'll do for a proof of concept.

Reply:
If we end up needing this a lot, we should switch to a feature flag imo.